### DISCUSSION TITLE
Auto history flush like fish shell

### DISCUSSION THREAD

**Discussion by wybert** [xonsh/xonsh#4932] [Q&A]
I am wondering how to configure xonsh to flush the history automatically after you close it like fish.

**Comment by anki-code** [comment]
By default the json history backend writes history after ending the session. And [SQLite history backend](https://xon.sh/tutorial_hist.html#sqlite-history-backend) writes the command to the DB on executing every command. I prefer and recommand using SQLite history backen because [the manipulation of the history](https://github.com/xonsh/xonsh/issues/2429#issuecomment-1235212003) is easy by using SQL language.

@wybert What do you mean by "flush" word? It could mean the writing (flushing) the history to the disk as well as skip writing.

**Reply by wybert** [reply]
I mean writing the history to the disk.

**Reply by anki-code** [reply]
By default the json history backend writes history after ending the session. And [SQLite history backend](https://xon.sh/tutorial_hist.html#sqlite-history-backend) writes the command to the DB on executing every command. I prefer and recommand using SQLite history backen because [the manipulation of the history](https://github.com/xonsh/xonsh/issues/2429#issuecomment-1235212003) is easy by using SQL language.
